dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -746,6 +746,41 @@ QueryProcessingStage::Enum InterpreterSelectQuery::executeFetchColumns(Pipeline
                     "[InterpreterSelectQuery::executeFetchColumns] no region query",
                     ErrorCodes::LOGICAL_ERROR);
         }
+        else
+        {
+            if (auto managed_storage = std::dynamic_pointer_cast<IManageableStorage>(storage); managed_storage)
```

Comment:
should we raise exception here if `managed_storage == nullptr`, just like above

```cpp
            TableID table_id = InvalidTableID;
            if (auto managed_storage = std::dynamic_pointer_cast<IManageableStorage>(storage); managed_storage)
            {
                table_id = managed_storage->getTableInfo().id;
            }
            else
            {
                throw Exception("Not supported request on non-manageable storage");
            }
```

Reply:
No, because there are some tests that run on other StorageEngine than `StorageDeltaMerge`.

https://github.com/pingcap/tiflash/blob/903525a3844c4be38fc17dcb34186557905d8d11/tests/delta-merge-test/example.test#L19-L26